### PR TITLE
Show completions for multiple modules if they share a name/alias

### LINKED ISSUE #405: Show completions from both modules when multiple imports have the same name/alias
When there are multiple imports with the same name or alias, the completions should show from both modules.

### PR DESCRIPTION
**Pull request by jmbockhorst**
Closes #405.

### CODE REVIEW

File: src/providers/astProvider.ts [hunk 1]
```diff
@@ -143,7 +143,7 @@ export class ASTProvider {
     );
 
     oldNodes.forEach((node) => {
-      if (!newIds.has(node.id)) {
+      if (node && !newIds.has(node.id)) {
```
Comment (jmbockhorst) [inline src/providers/astProvider.ts:146]: Was getting null nodes here that we weren't before. Maybe from the tree-sitter upgrade.

Comment (Razzeee) [reply src/providers/astProvider.ts:146]: Not sure how that's possible, it's defined in line 96. As `const oldNodes = tree?.rootNode.namedChildren ?? [];`.

File: test/completionProvider.test.ts [hunk 2]
```diff
@@ -1503,4 +1503,83 @@ func model =
...
+
+import Module as M
+
+func =
+    M.f{-caret-}
+`;
+
+    await testCompletions(
+      source,
+      ["foo", "Maybe", "Just", "Nothing"],
```
Comment (Razzeee) [inline test/completionProvider.test.ts:1527]: I think it's weird, that this case returns Maybe, Just etc as the caret is after the `f`

Comment (jmbockhorst) [reply test/completionProvider.test.ts:1527]: The client is responsible for filtering, this is how all our completions work.

File: src/providers/astProvider.ts (hunk 1 above)
Comment (jmbockhorst) [reply src/providers/astProvider.ts:146]: Tree sitter calls `delete` on some of those nodes?

Comment (Razzeee) [reply src/providers/astProvider.ts:146]: Might be, anyway, this check doesn't hurt.

Comment (jmbockhorst) [reply src/providers/astProvider.ts:146]: But it doesn't look like they do. All I know is that when it happened that array looked like `[Node, Node, ..., Node, null, null, null, ...]`

File: test/completionProvider.test.ts (hunk 2 above)
Comment (Razzeee) [reply test/completionProvider.test.ts:1527]: Right, forgot about that.

**Pull request merged**